File: CS/Form1.cs
using DevExpress.XtraPrinting;
using System;
using System.Linq;
using System.Windows.Forms;

namespace PrintGoToPage
{
    public partial class Form1 : Form
    {
        public delegate void ActivateFormDelegate();

        public void ActivateFormMethod() {
            this.Activate();
        }
        
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e) {
            for(int i = 0; i < 500; i++) {
                SimpleLink link = new SimpleLink("Simple Text " + i.ToString());
                compositeLink1.Links.Add(link);
            }
            
            compositeLink1.ShowPreview();
            button2.Enabled = true;
            button3.Enabled = true;

            this.BeginInvoke(new ActivateFormDelegate(ActivateFormMethod));
        }

        private void button2_Click(object sender, EventArgs e) {
            printingSystem1.PreviewFormEx.PrintControl.SelectedPageIndex = 2;
        }

        private void button3_Click(object sender, EventArgs e) {
            Page page = printingSystem1.Document.Pages[3];
            var brick = BrickSelector.GetBricks(page).Skip(4).FirstOrDefault();

            if(brick != null) {
                printingSystem1.PreviewFormEx.PrintControl.ShowBrick(brick, page);
                printingSystem1.MarkBrick(brick, page);
            }
        }

    }
}

File: VB/Form1.vb
Imports DevExpress.XtraPrinting
Imports System
Imports System.Linq
Imports System.Windows.Forms

Namespace PrintGoToPage
	Partial Public Class Form1
		Inherits Form

		Public Delegate Sub ActivateFormDelegate()

		Public Sub ActivateFormMethod()
			Me.Activate()
		End Sub

		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			If Not Me.button1.IsHandleCreated Then Return

			For i As Integer = 0 To 499
				Dim link As New SimpleLink("Simple Text " & i.ToString())
				compositeLink1.Links.Add(link)
			Next i

			compositeLink1.ShowPreview()
			button2.Enabled = True
			button3.Enabled = True

			Me.BeginInvoke(New ActivateFormDelegate(AddressOf ActivateFormMethod))
		End Sub

		Private Sub button2_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button2.Click
			If Not Me.button2.IsHandleCreated Then Return

			printingSystem1.PreviewFormEx.PrintControl.SelectedPageIndex = 2
		End Sub

		Private Sub button3_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button3.Click
			If Not Me.button3.IsHandleCreated Then Return

			Dim page As Page = printingSystem1.Document.Pages(3)
			Dim brick = BrickSelector.GetBricks(page).Skip(4).FirstOrDefault()

			If brick IsNot Nothing Then
				printingSystem1.PreviewFormEx.PrintControl.ShowBrick(brick, page)
				printingSystem1.MarkBrick(brick, page)
			End If
		End Sub

	End Class
End Namespace